src/replay_stage.rs

```diff
@@ -449,6 +433,40 @@ impl ReplayStage {
         let next_slots = blocktree.get_slots_since(&[slot_index]).expect("Db error");
         next_slots.first().cloned()
     }
+
+    fn forward_entries(
+        entries: &[Entry],
+        slot: u64,
+        slot_leader: Pubkey,
+        bank: &Arc<Bank>,
+        blocktree: &Arc<Blocktree>,
+        forward_entry_sender: &EntrySender,
+    ) -> Result<()> {
+        let parent_slot = blocktree.meta(slot).unwrap().map(|meta| meta.parent_slot);
+        let mut entry_tick_height = bank.tick_height();
+        let mut num_ticks_left_in_slot =
```

Comment:
`num_ticks_left_in_slot` is unnecessarily complicated. If you're not able to find the data you need from blocktree's metadata, you can at least change this to an incrementing `tick_index()` and then create EntryMeta with `num_ticks_left_in_slot: ticks_per_slot - tick_index`.  But better would be to change the interface with block explorer.  Instead of sending how many ticks are left, would it be sufficient if you sent only if it's the last tick?